libs/ui/src/search_select.tsx

```diff
@@ -0,0 +1,342 @@
+}: SearchSelectProps<T>): JSX.Element {
+  const [currentValues, setCurrentValues] = useState<T[]>(
+    defaultValue ? [defaultValue] : []
+  );
+  const currentValue = currentValues[0]; // only relevant value for single select
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [searchString, setSearchString] = useState<string>('');
+  const [isHidingInputCursor, setIsHidingInputCursor] = useState(false);
+  const [isShowingMenu, setIsShowingMenu] = useState<boolean>(false);
+
+  const optionLookup: Record<string, SelectOption> = {};
+  for (const option of options) {
+    optionLookup[option.value] = option;
+  }
+
+  function clearSearch() {
+    setSearchString('');
+  }
+
+  function focusInput() {
+    assert(inputRef.current);
+    inputRef.current.focus();
+  }
+
+  function resetOnFocus() {
+    clearSearch();
+    focusInput();
+  }
+
+  function resetOnBlur() {
+    clearSearch();
+    setIsShowingMenu(false);
+    setIsHidingInputCursor(false);
+  }
+
+  function onInput(e: React.ChangeEvent<HTMLInputElement>) {
+    setSearchString(e.target.value);
+    setIsHidingInputCursor(false);
+    setIsShowingMenu(true);
+  }
+
+  function updateCurrentValues(newValues: T[]) {
+    setCurrentValues(newValues);
+    onSelectionChange(newValues);
+  }
+
+  function onClickOption(value: T) {
+    if (isMulti) {
+      updateCurrentValues([...currentValues, value]);
+    } else if (value !== currentValue) {
+      updateCurrentValues([value]);
+      setIsHidingInputCursor(true); // hide input cursor after single selection
+    }
+    resetOnFocus();
+    setIsShowingMenu(false);
+  }
+
+  function onRemoveMultiValue(value: T) {
+    updateCurrentValues(currentValues.filter((v) => v !== value));
+  }
+
+  function onComponentBlur(e: React.FocusEvent<HTMLDivElement>) {
+    // when focus leaves the component
+    if (!e.currentTarget.contains(e.relatedTarget)) {
+      resetOnBlur();
+    }
+  }
+
+  const optionsMatchingSearch = options.filter((option) =>
+    option.label.toLowerCase().includes(searchString.toLowerCase())
+  );
+
+  const displayedOptions = isMulti
+    ? optionsMatchingSearch.filter(
+        (option) => !currentValues.includes(option.value)
+      )
+    : optionsMatchingSearch;
+
+  return (
+    <ComponentContainer onBlur={onComponentBlur}>
+      <SelectBar
+        onMouseDown={(e) => {
+          e.preventDefault(); // prevent blur within select bar
+          resetOnFocus();
+          setIsShowingMenu((show) => !show);
+          setIsHidingInputCursor(false);
+        }}
+      >
+        <ValuesContainer isMulti={isMulti}>
+          {isMulti &&
+            currentValues.map((value) => {
+              const option = optionLookup[value];
+              return (
+                <SelectedMultiValue key={option.value}>
+                  <SelectedMultiValueText>
+                    {option.label}
+                  </SelectedMultiValueText>
+                  <SelectedMultiValueRemove
```

Comment:
Since this is an icon-only button, worth assigning an `aria-label={Remove ${option label}}` - also makes it easier to write a more semantic test with `getButton("Remove foo")`, when combined with the note above about making this a `<button>` or `<div role="button">`.

Trying not to make assumptions about shared components only being used in a kiosk-browser context (where native web semantics don't matter as much), since we might have stuff like VxDesign hosted in a browser-accessible env

Reply:
handled by `react-select`, it's now a button and `aria-label` is generated by default